File: python/Problem003.py
# Problem 3
# 
# By: Tim Oram
#
# Runtime: 0.115s

import math
import sys

factors = []
number = 600851475143
sqrtnumber = int(math.floor(math.sqrt(number))) + 1

for i in xrange(2, sqrtnumber):
	# if number divides i
	if ((number % i) == 0):
		isPrime = True
		for j in factors:
			if ((i % j) == 0):
				isPrime = False
				break
		if isPrime:
			factors.append(i)

factors.sort()

print factors[-1]

File: c/Problem003.c
// Problem 2
// By: Tim Oram

#include <stdio.h>

int isPrime(unsigned long n) {
	
	int i;
	
	if ((n%2) == 0) {
		return 0;
	}
	
	for (i = 3; i*i < n; i +=2) {
		if ((n%i) == 0) {
			return 0;
		}
	}
	
	return 1;
	
}

int main(void) {
	
	const unsigned long number = 600851475143;
	
	unsigned long i, j;
	int is_prime = 0;
	
	unsigned long highest = 1;
	unsigned long number_of_primes = 2;
	
	for (i = 3; i*i <= number; i += 2) {
		if ((number%i) == 0 && isPrime(i) == 1) {
			highest = i;
		}
	}
	
	printf("Answer: %lu\n", highest);
	return 0;
}

